fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -5388,10 +5450,34 @@ class DDTeamCollectionUnitTest {
 		return Void();
 	}
 
+	ACTOR static Future<Void> AddTeamsBestOf_MultipleTeamSets() {
+		wait(Future<Void>(Void()));
+
+		Reference<IReplicationPolicy> policy = makeReference<PolicyAcross>(3, "zoneid", makeReference<PolicyOne>());
+		state int processSize = 6;
+		state int teamSize = 3;
+		state int teamSetCount = 3;
+		state std::unique_ptr<DDTeamCollection> collection =
+		    testTeamCollection(teamSize, policy, processSize, teamSetCount);
+
+		int desiredTeams = SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER * processSize;
+		int maxTeams = SERVER_KNOBS->MAX_TEAMS_PER_SERVER * processSize;
+
+		int result = collection->addTeamsBestOf(desiredTeams, desiredTeams, maxTeams);
+
+		ASSERT(collection->teamSets.size() == teamSetCount);
+		for (auto i = 0; i < collection->teamSets.size(); i++) {
+			ASSERT(collection->teamSets[i]->teamCount() >= result / teamSetCount);
```

Comment:
This assertion can pass even if `result=2` and all built teams are added to only one team set. An ASSERT for the min and max teamCount maybe better.

Reply:
I added an assertion that result > teamSetCount. That should have the same effect, i.e., the count is non-zero in addition to being similar.